Xamarin.Essentials/FilePicker/FilePicker.shared.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+
+namespace Xamarin.Essentials
+{
+    public static partial class FilePicker
+    {
+        public static Task<PickResult> PickFileAsync(PickOptions options = null)
+            => PlatformPickFileAsync(options);
+    }
+
+    public partial class PickOptions
+    {
+        public static PickOptions Default
+        {
+            get
+            {
+                return new PickOptions
+                {
+                    FileTypes = null,
+                };
+            }
+        }
+
+        public static PickOptions Images
+            => PlatformGetImagesPickOptions();
+
+        public string PickerTitle { get; set; }
+
+        public string[] FileTypes { get; set; }
```

Comment:
The file types are a bit tricky and platform dependent. On Android, you specify a list of MIME types, and you can also use wildcards, e.g. "image/png", or also "image/*"; default is "*/*". It also only grays out files that doesn't match the file types.

On iOS the values in UTType are used, which are not enum values, but iOS (and maybe version-) specific strings. There are values for very specific file types like .png, and "wildcard" values such as UTType.Image.

On UWP there's only file extension wildcards. So the pre-defined PickOptions.Default and PickOptions.Images options are there for specifying the appropriate FileTypes array values for the platform I'm running on.

You can see example code from the FilePicker plugin here:
https://github.com/jfversluis/FilePicker-Plugin-for-Xamarin-and-Windows/blob/master/samples/Plugin.FilePicker.Sample.Forms/Forms/MainPage.xaml.cs#L23

Reply:
@vividos @jamesmontemagno I think we need to have a look at this and maybe provide a slightly different way. Right now, it is going to be hard to use this api as it stands. For example, if we want to pick say a PDF file, the user would expect to maybe just do `*.pdf`. This will only work on UWP.

How would one restrict the selector to just PDF on iOS and Android (and maybe @rookiejava or @sung-su can chime in for Tizen).

Even if it is all very complicated, we can at least create a list of common types that is conditionally compiled:

```csharp
static class FilePickerFileTypes {
#if IOS
    public static string Png = UTType.Png;
#elif ANDROID
    public static string Png = "image/png"
#elif UWP
    public static string Png = "*.png"
#endif
}
```

We could also do one for all images, but then for UWP will have to have a big list of wildcards. Is a *.gif an image? what about *.bmp or *.webp?

NOTE: for this, we can't use constants as then they would get compiled into the dll, and not the reference we want.

As a summary, this is not really so cool if the user has to do this each time he/she want to select a picture from the gallery: https://github.com/jfversluis/FilePicker-Plugin-for-Xamarin-and-Windows/blob/master/samples/Plugin.FilePicker.Sample.Forms/Forms/MainPage.xaml.cs#L23-L42